src/libsystemd-network/dhcp6-option.c

```diff
@@ -285,24 +285,20 @@ int dhcp6_option_append_vendor_option(uint8_t **buf, size_t *offset, OrderedSet
 }
 
 static int option_append_ia_address(uint8_t **buf, size_t *offset, const struct iaaddr *address) {
-        struct iaaddr a;
-
         assert(buf);
         assert(*buf);
         assert(offset);
         assert(address);
 
         /* Do not append T1 and T2. */
-        a = (struct iaaddr) {
+        const struct iaaddr a = {
```

Comment:
So we can use const on function variables now?

Reply:
This is a structure… We pass it by reference to a function taking a bunch of pointers, some of them `void*`, so at least in principle somebody could pass the pointer in the wrong position and then the compiler would complain. I don't think that this falls under the umbrella of const function variables, because it has an "effect" outside of the function.

Reply:
Why not just inline the variable?

```
return dhcp6_option_append(buf, offset, SD_DHCP6_OPTION_IAADDR, sizeof(struct iaaddr), &(struct iaadr) { ... });
```

Reply:
We set two fields in the other place. Of course this *can* be inlined, but I I don't think it *should*. With the separate variable it's very clear what is assigned where, and inlining would give a very long line that wouldn't be very clear.

Reply:
I agree. Nevertheless, I think that the rule doesn't actually apply in this case.

Reply:
Hmm, what's the point of the "const" here? a local variable that is "const" makes little sense. That's just decoration without effect.

"const" really only makes sense on "static" stuff and on pointer types, but for local non-pointer variables it doesn't really make any sense because it is without effect. It doesn't declare a contract to the outside, for interaction with other code. or what am I missing?

I wouldn't put noise here that has no effect, I must say. it just raises eyebrows when reading through the code.

Reply:
Please see my answer to the *this very question* right above.

Reply:
> Hmm, what's the point of the "const" here? a local variable that is "const" makes little sense. That's just decoration without effect.

lol. The effect is quite literally to prevent accidental modification as well as working as a documentation to future users that modifying this variable it not intended. It has its uses.

Reply:
Yes, to catch stupid mistakes and to guide the reader.